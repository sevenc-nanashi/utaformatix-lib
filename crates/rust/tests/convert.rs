extern crate utaformatix_rs;

use duplicate::duplicate_item;
use tracing_test::traced_test;
use utaformatix_rs::{GenerateOptions, ParseOptions};

#[rstest::fixture]
fn utaformatix() -> utaformatix_rs::base::UtaFormatix {
    utaformatix_rs::base::UtaFormatix::new()
}

#[duplicate_item(
    test_name                 function             path;
    [test_parse_standard_mid] [parse_standard_mid] ["generated/standard.mid"];
    [parse_music_xml]         [parse_music_xml]    ["generated/musicXml.musicxml"];
    [parse_ccs]               [parse_ccs]          ["generated/cevio.ccs"];
    [parse_dv]                [parse_dv]           ["generated/dv.dv"];
    [parse_ustx]              [parse_ustx]         ["generated/openutau.ustx"];
    [parse_svp]               [parse_svp]          ["generated/synthV.svp"];
    [parse_tssln]             [parse_tssln]        ["voisona.tssln"];
    [parse_uf_data]           [parse_uf_data]      ["generated/ufdata.ufdata"];
    [parse_vocaloid_mid]      [parse_vocaloid_mid] ["generated/vocaloid.mid"];
    [parse_vsq]               [parse_vsq]          ["generated/vsq.vsq"];
    [parse_vsqx]              [parse_vsqx]         ["generated/vsqx.vsqx"];
    [parse_vpr]               [parse_vpr]          ["generated/vpr.vpr"];
)]
#[rstest::rstest]
#[tokio::test]
#[traced_test]
async fn test_name(utaformatix: utaformatix_rs::base::UtaFormatix) {
    let data = include_bytes!(concat!("../utaformatix-ts/testAssets/", path));
    let options = ParseOptions::default();
    let result = utaformatix.function(data, options).await;

    let parsed = result.expect("Failed to parse data");

    insta::assert_debug_snapshot!(parsed);
}

#[duplicate_item(
    test_name                 function             path;
    [test_parse_ust]          [parse_ust]          ["generated/utau.ust"];
)]
#[rstest::rstest]
#[tokio::test]
#[traced_test]
async fn test_name(utaformatix: utaformatix_rs::base::UtaFormatix) {
    let data = include_bytes!(concat!("../utaformatix-ts/testAssets/", path));
    let options = ParseOptions::default();
    let result = utaformatix.function(&[data], options).await;

    let parsed = result.expect("Failed to parse data");

    insta::assert_debug_snapshot!(parsed);
}

#[duplicate_item(
    test_name                    function;
    [test_generate_standard_mid] [generate_standard_mid];
)]
#[rstest::rstest]
#[tokio::test]
#[traced_test]
async fn test_name(utaformatix: utaformatix_rs::base::UtaFormatix) {
    let data = include_bytes!("../utaformatix-ts/testAssets/generated/standard.mid");
    let ufdata = utaformatix
        .parse_standard_mid(data, ParseOptions::default())
        .await
        .expect("Failed to parse data");
    let options = GenerateOptions::default();
    let result = utaformatix.function(ufdata, options).await;

    result.expect("Failed to generate data");
}
