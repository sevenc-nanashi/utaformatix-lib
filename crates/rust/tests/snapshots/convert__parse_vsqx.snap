---
source: crates/rust/tests/convert.rs
expression: parsed
---
UfData {
    format_version: 1,
    project: Project {
        name: "data",
        tracks: [
            Track {
                name: "Track",
                notes: [
                    Note {
                        key: 69,
                        tick_on: 240,
                        tick_off: 720,
                        lyric: "„Å©",
                        phoneme: Some(
                            "a",
                        ),
                    },
                ],
                pitch: None,
            },
        ],
        time_signatures: [
            TimeSignature {
                measure_position: 0,
                numerator: 4,
                denominator: 4,
            },
        ],
        tempos: [
            Tempo {
                tick_position: 0,
                bpm: 120,
            },
        ],
        measure_prefix: 4,
    },
}
